templates/default/bump_environments.rb.erb

```diff
@@ -44,8 +51,16 @@ Git.configure do |config|
 end
 git = Git.open('.')
 git.branch('master').checkout
-chef_envs_string = all_envs ? 'all-envs' : chef_envs.join(',')
-git_branch = "jenkins/bump-#{cookbook}-to-#{version}-in-#{chef_envs_string}"
+branch_hash = if is_all_envs
+                'in-all-envs'
+              elsif is_default_envs
+                'in-default-envs'
+              else
+                ''
+              end
+# Add a 10-digit hash of the version and envs so the branch names are unique.
+branch_hash += "-#{(chef_envs.to_a << version).join(',').hash % 10_000_000_000}"
```

Comment:
```
1_000_000.times do
  "#{47589468927645687 % 10_000_000_000}"
end

real	0m0.497s
```
```
1_000_000.times do
  "#{47589468927645687.to_s.slice(0, 10)}"
end

real	0m0.612s
```

Modding seems to be faster, and the code is about the same in length and complexity, so I vote to keep it how it is.  I'll make it a little shorter though, sure.